src/runner/index.ts

```diff
@@ -110,31 +110,41 @@ export class MainRunner extends Runner {
             return true;
         }
 
-        const collection = TestCollection.create({ [browserId]: [test] });
-        this.running.add(this._runTestsInBrowser(collection, browserId));
+        const browserRunner = this._addTestToBrowserRunner(test, browserId);
+        this.running.add(this._waitBrowserRunnerTestsCompletion(browserRunner));
 
         return true;
     }
 
     protected async _runTests(testCollection: TestCollection): Promise<void> {
-        testCollection.getBrowsers().forEach((browserId: string) => {
-            this.running.add(this._runTestsInBrowser(testCollection, browserId));
-        });
+        testCollection.eachTestAcrossBrowsers((test, browserId) => this._addTestToBrowserRunner(test, browserId));
+
+        this.activeBrowserRunners.forEach(runner => this.running.add(this._waitBrowserRunnerTestsCompletion(runner)));
 
         return this.running.done();
     }
 
-    protected async _runTestsInBrowser(testCollection: TestCollection, browserId: string): Promise<void> {
+    protected _addTestToBrowserRunner(test: Test, browserId: string): BrowserRunner {
+        const browserRunner = this.activeBrowserRunners.get(browserId) || this._createBrowserRunner(browserId);
+        browserRunner.addTestToRun(test);
```

Comment:
Now all tests are added to `BrowserRunner` using method `addTestToRun`, because I need to add tests one by one, as a browser request is being executed under the hood. `MainRunner` (this file) -> `BrowserRunner` -> `RegularTestRunner` (here request for browser)